### PR TITLE
Tests expecting NotFound response now look for generic error message

### PR DESCRIPTION
**Pull request by donnie-msft**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1313

Regression? Last working version: n/a

## Description

These tests just need to determine if an HTTP error occurred at all. A server may return a variety of specific error codes and messages, and there's no benefit to depending on that.

(Also made some coding guideline refactoring)

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added - updated a test
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Xunit;
+
+namespace NuGet.Common.Test
+{
+    public class DateTimeUtilityTests
```
Comment (zivkan) [inline test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs]: This class looks unrelated the issue.

Comment (donnie-msft) [reply test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs]: I have no idea what's going on. None of these are files I changed....

> git diff dev --name-only
> test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetListCommandTest.cs
> test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPushCommandTest.cs

Comment (donnie-msft) [reply test/NuGet.Core.Tests/NuGet.Common.Test/DateTimeUtilityTests.cs]: There we go, must have been a bad rebase somehow.

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetListCommandTest.cs [hunk 2]
```diff
@@ -921,26 +921,26 @@ public void ListCommand_InvalidInput_V3_NonExistent(string invalidInput)
...
-                    result.Item1 != 0,
-                    "The run did not fail as desired. Simply got this output:" + result.Item2);
+                Assert.False(
+                    result.Success,
+                    "The run did not fail as desired. Simply got this output:" + result.Output);
 
                 Assert.True(
-                    result.Item3.Contains("400 (Bad Request)"),
-                    "Expected error message not found in " + result.Item3
+                    result.Errors.Contains("Response status code does not indicate success"),
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetListCommandTest.cs:942]: Actual change here

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPushCommandTest.cs [hunk 3]
```diff
@@ -2024,20 +2024,20 @@ public void PushCommand_InvalidInput_V3_NotFound(string invalidInput)
...
-                    result.Item1 != 0,
-                    "The run did not fail as desired. Simply got this output:" + result.Item2);
+                Assert.False(
+                    result.Success,
+                    "The run did not fail as desired. Simply got this output:" + result.Output);
 
                 Assert.True(
-                    result.Item3.Contains("400 (Bad Request)"),
-                    "Expected error message not found in " + result.Item3
+                    result.Errors.Contains("Response status code does not indicate success"),
```
Comment (donnie-msft) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:2039]: Actual change here

**Review by zivkan — APPROVED** [review]

**Review by dominoFire — APPROVED** [review]
